truss-chains/truss_chains/stub.py

```diff
@@ -1,20 +1,38 @@
 import abc
 import asyncio
+import contextlib
+import contextvars
 import logging
 import ssl
 import threading
 import time
-from typing import Any, ClassVar, Mapping, Optional, Type, TypeVar, final
+from typing import Any, ClassVar, Iterator, Mapping, Optional, Type, TypeVar, final
 
 import aiohttp
 import httpx
+import starlette.requests
 import tenacity
 
 from truss_chains import definitions, utils
 
 DEFAULT_MAX_CONNECTIONS = 1000
 DEFAULT_MAX_KEEPALIVE_CONNECTIONS = 400
 
+_trace_parent_context: contextvars.ContextVar[str] = contextvars.ContextVar(
+    "trace_parent"
+)
+
+
+@contextlib.contextmanager
+def trace_parent(request: starlette.requests.Request) -> Iterator[None]:
+    token = _trace_parent_context.set(
```

Comment:
If `_trace_parent_context` is global are there concurrency implications on parallel requests?